pandas/core/reshape/pivot.py

```diff
@@ -1163,9 +1163,15 @@ def _normalize(
 
     elif margins is True:
         # keep index and column of pivoted table
+
         table_index = table.index
         table_columns = table.columns
-        last_ind_or_col = table.iloc[-1, :].name
+        try:
+            last_ind_or_col = table.iloc[-1, :].name
+        except IndexError as err:
+            raise IndexError(
+                "Can't get margins since the result dataframe is empty"
```

Comment:
It's not clear to me that this is an improvement in the error message. For example with the OP of the linked issue, isn't a natural question a user would have is "Why is the result DataFrame empty?"